core/src/test/java/org/elasticsearch/index/mapper/MapperServiceTest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.test.ElasticsearchSingleNodeTest;
+import org.junit.Test;
+
+public class MapperServiceTest extends ElasticsearchSingleNodeTest {
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Instead of using expected exception, cant eh test wrap the call in a try/catch, and actually check the exception is what we think it should be? Something like:
```
try {
    ...prepareIndex...
} catch (IllegalArgumentException e) {
    assertTrue(e.getMessage().contains("must not start with a '.'"));
}
```

Also I think it would be better to use an explicit create index call with mappings here, instead of relying on dynamic mappings?